test/e2e/kubectl.go

```diff
@@ -608,11 +608,40 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 			framework.Logf("copying %s to the %s pod", kubectlPath, simplePodName)
 			framework.RunKubectlOrDie("cp", kubectlPath, ns+"/"+simplePodName+":/tmp/")
 
+			// Build a kubeconfig file that will make use of the injected ca and token,
+			// but point at the DNS host and the default namespace
+			iccOverride, err := ioutil.TempFile("", "icc-override.kubeconfig")
+			framework.ExpectNoError(err)
+			defer func() { os.Remove(iccOverride.Name()) }()
+			framework.ExpectNoError(ioutil.WriteFile(iccOverride.Name(), []byte(`
+kind: Config
+apiVersion: v1
+clusters:
+- cluster:
+    api-version: v1
+    server: https://kubernetes.default.svc:443
+    certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+  name: kubeconfig-cluster
+contexts:
+- context:
+    cluster: kubeconfig-cluster
+    namespace: default
+    user: kubeconfig-user
+  name: kubeconfig-context
+current-context: kubeconfig-context
+users:
+- name: kubeconfig-user
+  user:
+    tokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+`), os.FileMode(0755)))
+			framework.Logf("copying %s to the %s pod", iccOverride.Name(), simplePodName)
+			framework.RunKubectlOrDie("cp", iccOverride.Name(), ns+"/"+simplePodName+":/tmp/icc-override.kubeconfig")
```

Comment:
I am not sure if this is a regression, but it appears you can't change the destination filename; i.e. `kubectl cp sourcefile pod:/tmp/destfile` will yield `/tmp/sourcefile` inside the container :-/ I discovered this a few days ago when testing something else.